src/hooks/useLogin.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+
+const HTTP_OK = 200;
+const HTTP_UNAUTHORIZED = 401;
+
+function useLogin() {
+    const API_URL = import.meta.env.VITE_API_URL;
+
+    const [loading, setLoading] = useState<boolean>(true);
+    const [loggedIn, setLoggedIn] = useState<boolean>(false);
+
+    useEffect(() => {
+        const getUserData = async () => {
+            const response = await fetch(API_URL + "/dashboard/login", {
+                credentials: "include",
+            });
+
+            switch (response.status) {
+            case HTTP_OK:
```

Comment:
if you (want) use Prettier you should run it so that you dont have a huge commit some time in future and wrong commit messages